lib/std/start.zig

```diff
@@ -329,13 +329,23 @@ fn _start() callconv(.Naked) noreturn {
             \\ jsr (%%pc, %%a0)
             ,
             .mips, .mipsel =>
+            \\ bal 1f
+            \\ .gpword .
+            \\ 1:
+            \\ lw $gp, 0($ra)
+            \\ subu $gp, $ra, $gp
```

Comment:
The `gp` (global pointer) register serves a somewhat similar purpose to `r2` (ToC pointer) on powerpc; it's required for dynamically-linked binary to work.